extract/match_spacy_w_wrgl_uid.py

```diff
@@ -0,0 +1,71 @@
+import sys
+
+sys.path.append("../")
+import pandas as pd
+import os
+from datamatch import JaroWinklerSimilarity, ThresholdMatcher, ColumnsIndex
+from lib.path import data_file_path
+from lib.clean import clean_names
+import glob
+
+
+#  match appeal_uids with event_uids
```

Comment:
@pckhoi the matching runs very slowly. Is it due to what you were referencing recently regarding the added benefit of sets?

Reply:
It doesn't look like you're dealing with sets here. You only deal with sets if each row contains a finite number of unordered elements from an also finite set, which was the case for matching uids. First name and last name are definitely not sets in any sense. They are strings which are ordered lists of characters. You cannot use set operations on strings. String similarity functions such as Jaro Winkler's and Levenshtein's  are the ideal way to compare strings. Looks like you're already using one.

You have to look at how many pairs you're comparing and how many rows it is comparing per second. Both infos are available if you set `show_progress=True`. If the number of pairs is too high then you need a new index that is even more discrimative. If you don't know what other criteria that allow a pair to be matched but you do know other criteria that eliminate a pair then you can add a filter.

That and if you're only planning to match pairs perfectly (decision = 1) then there are little point in using datamatch. Just use pandas filtering instead which is guaranteed to be faster.

Reply:
Thanks for the feedback!